lib/xcode.js

```diff
@@ -136,20 +128,20 @@ const getVersionMemoized = _.memoize(
 );
 
 async function getVersion (parse = false, retries = DEFAULT_NUMBER_OF_RETRIES, timeout = XCRUN_TIMEOUT) {
-  let version = await getVersionMemoized(retries, timeout);
+  const version = await getVersionMemoized(retries, timeout);
+  // xcode version strings are not exactly semver string: patch versions of 0
+  // are removed (e.g., '10.0.0' => '10.0')
+  const versionString = version.patch > 0 ? version.version : `${version.major}.${version.minor}`;
   if (!parse) {
-    return version;
+    return versionString;
   }
-  let match = /^(\d+)\.(\d+)(\.(\d+))?$/.exec(version);
-  // match should be an array, either of
-  //     [ '7.0', '7', '0', undefined, undefined, index: 0, input: '7.0' ]
-  //     [ '7.0.1', '7', '0', '.1', '1', index: 0, input: '7.0.1' ]
+
   return {
-    versionString: version,
-    versionFloat: parseFloat(`${match[1]}.${match[2]}`),
-    major: parseInt(match[1], 10),
-    minor: parseInt(match[2], 10),
-    patch: match[4] ? parseInt(match[4], 10) : undefined
+    versionString,
+    versionFloat: parseFloat(versionString),
```

Comment:
`parseFloat(''10.0.1'') = 10.0 = parseFloat(''10.0.2'')` - is it what we want to get?